Content/Items/Currency/AscendantShard.cs

```diff
@@ -0,0 +1,48 @@
+﻿using PathOfTerraria.Common.Enums;
+using PathOfTerraria.Core.Items;
+using Terraria.ID;
+
+namespace PathOfTerraria.Content.Items.Currency;
+
+internal class AscendantShard : CurrencyShard
+{
+	public override void SetDefaults()
+	{
+		base.SetDefaults();
+		Item.rare = ItemRarityID.Blue;
+	}
+
+	public override bool CanRightClick()
+	{
+		Item heldItem = Main.LocalPlayer.HeldItem;
+		if (!heldItem.TryGetGlobalItem(out PoTGlobalItem _))
+		{
+			return false;
+		}
+		
+		if (heldItem.GetInstanceData().Rarity != ItemRarity.Magic &&
+		    heldItem.GetInstanceData().Rarity != ItemRarity.Rare)
+		{
+			Main.NewText("This item can only be used on Magic or Rare items.");
+			return false;
+		}
+
+		if (PoTItemHelper.HasMaxAffixesForRarity(heldItem))
+		{
+			Main.NewText("This item has maximum affixes already");
+			return false;
+		}
+		
+		return base.CanRightClick();
+	}
+
+	public override void RightClick(Player player)
+	{
+		if (player.HeldItem == null)
```

Comment:
This can never be run as CanRightClick's first action is assuming HeldItem is not null. If it's null, it'll throw before this.

Reply:
Ok, you have this in several files. Did you add this for a specific reason? This should be a check in CanRightClick if anything, but I don't think this should be possible as the player *always* holds an item, it's just usually empty.

Reply:
I was noticing that if you right clicked the item, without anything in your hands, you would get a null reference exception.

Reply:
Huh, that's bizarre. Definitely move it up to CanRightClick, but that shouldn't be possible if I understand the game right. Are you sure HeldItem is the null object?

Reply:
No clue, launched the game again and wasn't seeing it anymore. 
Resolved here: 9e4053464e74ad73dac24295830b71fa0131c1dd
That said I am noticing it requires and consumes two orbs before the effect is added atm. Unsure if my changes or is just a current bug